lib/postgrex/extensions/network.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Postgrex.Extensions.Network do
+  @moduledoc false
+  alias Postgrex.TypeInfo
+  use Postgrex.BinaryExtension,
+    [send: "inet_send", send: "cidr_send", send: "macaddr_send"]
+
+  def encode(%TypeInfo{type: "inet"}, %Postgrex.INET{} = inet, _, _),
+    do: encode_network(inet)
+  def encode(%TypeInfo{type: "cidr"}, %Postgrex.CIDR{} = cidr, _, _),
+    do: encode_network(cidr)
+  def encode(%TypeInfo{type: "macaddr"}, %Postgrex.MACADDR{} = macaddr, _, _),
+    do: encode_network(macaddr)
```

Comment:
This example is for encode, not decode, so didn't think was worth it as encoding is usually a few parameters and decoding is many rows. The decode code does not use type info.

Great you mentioned this though, macaddr technically needs to be moved to is own extension because it could conflicts with an inet ipv4 address.